service/web/tabs/app/src/components/ServicesListComponent.tsx

```diff
@@ -3,6 +3,26 @@ import { ErrorCalloutComponent } from "@misk/core"
 import * as React from "react"
 import { Link } from "react-router-dom"
 
+export interface IServiceLinkProps {
+  name: string
+  variants: (string | null)[]
+}
+
+function ServiceLink(props: IServiceLinkProps) {
+  if (props.variants.length > 1) {
+    return (
+      <Link to={`/app/services/${props.name}/variants/`}>
+        {this.props.name}
+      </Link>
+    );
+  }
+  return (
+    <Link to={`/app/services/${props.name}/variants/${props.variants[0]}`}>
```

Comment:
if there's only one variant, I wouldn't change anything about how the UI works. This looks like it will break old urls.

Reply:
As a first pass, I've left this piece of code as-is.

Behaviorally, this should be the same as the old UI. If there's a single variant, the UI will proceed directly to the original page. If there are multiple, it takes the user to a subpage listing out variants.

For old urls, I've added redirects that will send the user to the new page, using `RESERVED_VARIANT` for the dynamic segment.